sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -452,6 +452,7 @@ class RocksDB(
    * Drop uncommitted changes, and roll back to previous version.
    */
   def rollback(): Unit = {
+    acquire()
```

Comment:
nit: This is now a side-effect change. Do we want to keep this or just remove this? I feel like it's safer to add acquire() here, but once you update the PR title and description to reflect the new direction, this change is beyond the scope.

Reply:
This is actually not exactly related to this change. But we had missed this call for this function. Do you prefer to create this in a separate change ?

Reply:
Discussed with @HeartSaVioR - this is broken. but will fix this in a separate PR